### DISCUSSION TITLE
Dwa Arduino Mega do Maliny po USB - czy to możliwe?

### DISCUSSION THREAD

**Discussion by Mesztica** [lkankowski/arduino-multi-relay2#46] [Q&A]
Hej,
skecz u mnie śmiga od 2 lat bez żadnych problemów - podłączony po USB do Maliny i zintegrowany z HA przez MySensors. Sęk w tym że właśnie uruchomiłem piętro domu i tam mam drugie Mega (w sumie mam 65 obwodów oświetlenia i 75 łączników). Za cholerę tego nie mogę skonfigurować. Podłączenie do Maliny wykrywa drugie identyczne urządzenie i generalnie żadne nie działa. Oczywiście MULTI_RELAY_DESCRIPTION ustawione unikatowe i oczywiście wszystkie ID przekaźników również unikatowe w skali obu urządzeń. Czy to w ogóle możliwe żeby mieć więcej niż jedno Arduino z tym skeczem? Pozdrawiam

**Comment by lkankowski** [comment]
Zacznij może od tego, że oba Arduino Mega podłączone przez USB są widoczne w systemie?
`lsusb` powinno Ci pomóc.
Jeżeli oba widoczne, to czy są prawidłowo widoczne w HA?

**Comment by Mesztica** [comment]
Dziękuję za szybką odpowiedź :)
No i właśnie już tutaj zauważyłem problem, ale niestety moja umiejętność zarządzania systemem linuxowym to jakieś 1 na 10.

Ale doszedłem do tego, że pierwsze Arduino jest dostępne pod ttyUSB0, po podłączeniu drugiego Adruino (i restarcie maliny) pojawia się ttyUSB1, które jednak ma identyczny identyfikator co pierwsze Arduino, czyli: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
W HA zauważyłem pojawienie się duplikatów encji.

Zmieniałem porty USB na malinie, ale efekt ten sam. Po odłączeniu nowego Arduino wszystko wraca do normy w HA a cały czas oba Arduino autonomicznie funkcjonują bez zastrzeżeń z fizycznymi łącznikami.

**Reply by lkankowski** [reply]
Masz oryginalne Arduino Mega czy klona?
Spróbuj w necie poszukać jak odpalić 2 USB z tym samym ID. Możesz też popróbować z połączeniem przez hub, co ogólnie i tak jest dobrym pomysłem, bo u mnie na Raspberry 3B+ czasami nie chce bootować jak nie jestem podpięty przez hub.

**Reply by Mesztica** [reply]
Nie jestem pewien, ale skoro kupiłem hurtowo za 3 lata temu to pewnie były klony po ok 50zł.
Czy podłączenie przez hub może automatycznie rozwiązać problem? Tzn urządzenia dostaną unikatowe identyfikatory? A jaki hub byś polecił do tego celu?

**Reply by lkankowski** [reply]
Jak po 50 zł, to na bank klony. Jak nie masz skila technicznego, to możliwe, że najszybciej będzie kupić choć jedno oryginalne, bo one chyba mają unikalne ID (nie dam sobie za to ręki uciąć).

A jak chcesz diagnozować problem, to można zacząć od polecenia `lsusb` w każdym wariancie:
1. bez podłączonego Arduino
2. z podłączonym Arduino nr 1
3. z podłączonym Arduino nr 2
4. z podłączonymi oboma Arduino

Jak chcesz ułatwić analizę, to wywal z wyników 2, 3 i 4 wszystko co zwrócił wariant 1.

PS.
Szybkie googlowanie dało jeszcze jedną podpowiedź - spróbuj wyłączyć serial na GPIO (`sudo raspi-config` a następnie: "Interface Options" > "P6 Serial Port" > "No" lub "Disable").
I jeszcze jakieś podpowiedzi mogą być tu:
- https://forums.raspberrypi.com/viewtopic.php?t=164935

**Reply by Mesztica** [reply]
Spróbuję najpierw to odpalić przez Ethernet tylko muszę dołożyć W5100 co będzie mnie kosztowało kilka wejść. Ale to może być dobre rozwiązanie bo i tak chcę zmigrować z maliny na jakiegoś PCta i wtedy połączenie USB nie będzie możliwe ze względu na inną lokalizację serwera.

Póki co nie wiem jak obsłużyć W5100 w skeczu, ale poszukam na necie i pewnie się da :)
Dam sobie chwilę czasu i na pewno dam potem znać jak poszło.
